encoding/encoder_types.go

```diff
@@ -162,6 +162,20 @@ func (se *structEncoder) encode(v reflect.Value) interface{} {
 			encField = getReferenceField(f, v, encField)
 		}
 
+		if f.compound {
+			compoundField, ok := m[f.name].([]interface{})
+			if !ok {
+				compoundField = make([]interface{}, f.compoundIndex+1)
+			} else if len(compoundField) < f.compoundIndex+1 {
+				tmp := make([]interface{}, f.compoundIndex+1)
```

Comment:
To avoid reallocation, would it be worth setting a max number of compound components, like 10, and then always allocating a slice with capacity for that many elements?